Aplib.Core/Desire/Tactic.cs

```diff
@@ -0,0 +1,17 @@
+namespace Aplib.Core.Desire
+{
+    /// <summary>
+    /// Tactics are the real meat of <see cref="Goal"/>s, as they define how the agent can approach the goal in hopes
+    /// of finding a solution which meets the Goal's predicate. A tactic represents a smart combination of
+    /// <see cref="Action"/>s, which are executed in a Believe Desire Intent Cycle.
+    /// </summary>
+    /// <seealso cref="Goal"/>
+    /// <seealso cref="Action"/>
+    public abstract class Tactic
+    {
+        /// <summary>
+        /// Execute the next cycle in the Believe Desire Intent Cycle.
+        /// </summary>
+        public abstract void IterateBdiCycle();
```

Comment:
Maybe my personal preference, but capitalize BDI since we capitalize the start of all words, like IterateBDICycle()? Or don't we do this because it is an afkorting?

Reply:
According to this [SO answer](https://stackoverflow.com/questions/2236807/java-naming-convention-with-acronyms), if there is an abreviation of three letters or more, this is well-named.

> The .NET Framework guidelines recommend that acronyms three letters or longer use mixed case for PascalCase and camelCase identifiers, except for the first word of a camelCase identifier. Thus for a class name you might have XmlDocument, while a local variable might be named httpRequest

Reply:
It's cursed, but code warnings are more cursed. Rider doesn't like names like `CustomerID` either...

Reply:
Then it is fine!